### PR TITLE
Don't worry about unsed lines for some functions.

### PR DESCRIPTION
**Pull request by guy-maurel**
These functions are only used at developement time.

### CODE REVIEW

File: src/logmask.cpp [hunk 1]
```diff
@@ -12,6 +12,14 @@
 #include "unc_ctype.h"
 
 
+/*
+ * the test suite Coveralls: https://coveralls.io
+ * will complains because these functions are only
+ * used at developement time.
+ * Don't worry about unsed lines for the functions:
+ *   logmask_to_str
+ *   logmask_from_string
```
Comment (mwoehlke-kitware) [inline src/logmask.cpp]: Really? It seems `logmask_from_string` should be called any time `-L` is used, and we *definitely* have tests for that... (`logmask_to_str` OTOH is totally unused. How likely is it to actually be useful?)

File: src/logmask.h [hunk 2]
```diff
@@ -14,6 +14,14 @@
 #include <bitset>
 #include "log_levels.h"
 
+/*
+ * the test suite Coveralls: https://coveralls.io
+ * will complains because these functions are only
+ * used at developement time.
+ * Don't worry about unsed lines for the functions:
+ *   logmask_test
+ *   logmask_set_all
```
Comment (mwoehlke-kitware) [inline src/logmask.h]: Similarly, it seems like both of these should be exercised?

**Comment by mihaipopescu** [comment]
As @mwoehlke-kitware was saying, cli should be part of the coverage. Try wrapping them in some defines that are set only in dev builds and see if the build succeeds.

File: src/logmask.h (hunk 2 above)
Comment (mwoehlke-kitware) [reply src/logmask.h]: Okay, I see you used the PP to hide the devel-only stuff :slightly_smiling_face:. However, these are not hidden, and I believe they *are* being exercised. Thus, is this comment (re: `logmask_test` and `logmask_set_all`) correct? I think it is not?

**Comment by mihaipopescu** [comment]
I'm still confused where this functions are used... but ok, if `ifdef`ing them makes everyone happy it's cool.
I wonder if at any point you would like to make a build with this define set to add a new cmake config.. or maybe debug should always define that but not when coverage is set.

**Comment by mwoehlke-kitware** [comment]
> I'm still confused where this functions are used...

They "aren't". They apparently exist so that someone hacking on uncrustify can use them to help with debugging while hacking, but aren't intended to be used in checked-in code. Using the PP to hide them serves as a clever trick to tell coverage to ignore their existence, which (IMHO) makes sense, given the previous sentence.

**Pull request merged**